src/views/domain-page/domain-page-cluster-selector/domain-page-cluster-selector.tsx

```diff
@@ -0,0 +1,49 @@
+'use client';
+import React from 'react';
+import { Select, SIZE } from 'baseui/select';
+import { useRouter, usePathname } from 'next/navigation';
+
+import { overrides, styled } from './domain-page-cluster-selector.styles';
+import { DomainHeaderInfoItemContentProps } from '../domain-page-header-info/domain-page-header-info.types';
+
+export default function DomainPageClusterSelector(
+  props: DomainHeaderInfoItemContentProps
+) {
+  const router = useRouter();
+  const currentPath = usePathname();
+
+  if (props.domainInfo.clusters?.length === 1) {
+    return <styled.ItemLabel>{props.cluster}</styled.ItemLabel>;
+  }
+
+  return (
+    <Select
+      overrides={overrides.select}
+      options={props.domainInfo.clusters.map((cluster) => ({
+        id: cluster.clusterName,
+        label: cluster.clusterName,
+      }))}
+      value={[
+        {
+          id: props.cluster,
+          label: props.cluster,
+        },
+      ]}
+      onChange={({ option }) => {
+        if (option && option.id !== props.cluster) {
+          const newPath = currentPath.replace(
```

Comment:
This line is fragile and can be improved. We don't have any limitations for naming clusters to match any other part of the URL. Also, it's better to make it more clear which url this component is navigating to instead of manipulating the current url.

FYI: https://nextjs.org/docs/app/building-your-application/configuring/typescript#statically-typed-links